format.go

```diff
@@ -98,33 +98,33 @@ type protobufFormat struct{}
 
 // NewDecoder implements Format.
 func (protobufFormat) NewDecoder(r io.Reader) (FormatDecoder, error) {
-	return protobufWrapper{Reader: pbio.NewDelimitedReader(r, 1<<28)}, nil
+	return protobufWrapper{dec: pbio.NewDecoder(r)}, nil
 }
 
 // NewEncoder implements Format.
 func (protobufFormat) NewEncoder(w io.Writer) (FormatEncoder, error) {
-	return protobufWrapper{Writer: pbio.NewDelimitedWriter(w)}, nil
+	return protobufWrapper{enc: pbio.NewEncoder(w)}, nil
 }
 
 type protobufWrapper struct {
-	pbio.Reader
-	pbio.Writer
+	dec *pbio.Decoder
+	enc *pbio.Encoder
 }
 
 func (w protobufWrapper) Decode(v interface{}) error {
 	msg, ok := v.(proto.Message)
```

Comment:
same here, instead of a single cast, you can make a switch and support both gogo + new pb